code/game/objects/items/crayons.dm

```diff
@@ -823,31 +820,34 @@
 		if(ishuman(carbon_target) && actually_paints)
 			var/mob/living/carbon/human/human_target = carbon_target
 			human_target.update_lips("spray_face", paint_color)
-		. = use_charges(user, 10, FALSE)
+		use_charges(user, 10, FALSE)
 		var/fraction = min(1, . / reagents.maximum_volume)
 		reagents.expose(carbon_target, VAPOR, fraction * volume_multiplier)
 
-		return .
+	else if(actually_paints && target.atom_colours[atom_colours] == paint_color)
+		balloon_alert(user, "[target.p_theyre()] already of that color.")
+		return FALSE
 
 	if(ismob(target) && (HAS_TRAIT(target, TRAIT_SPRAY_PAINTABLE)))
 		if(actually_paints)
 			target.add_atom_colour(paint_color, WASHABLE_COLOUR_PRIORITY)
 			SEND_SIGNAL(target, COMSIG_LIVING_MOB_PAINTED)
-		. = use_charges(user, 2, requires_full = FALSE)
+		use_charges(user, 2, requires_full = FALSE)
 		reagents.trans_to(target, ., volume_multiplier, transfered_by = user, methods = VAPOR)
 
 		if(pre_noise || post_noise)
 			playsound(user.loc, 'sound/effects/spray.ogg', 5, TRUE, 5)
 		user.visible_message(span_notice("[user] coats [target] with spray paint!"), span_notice("You coat [target] with spray paint."))
-		return .
+		return
 
 
 	if(isobj(target) && !(target.flags_1 & UNPAINTABLE_1))
+		var/color_is_dark = FALSE
 		if(actually_paints)
-			var/color_is_dark = is_color_dark(paint_color)
+			color_is_dark = is_color_dark(paint_color)
 
 			if (color_is_dark && !(target.flags_1 & ALLOW_DARK_PAINTS_1))
-				to_chat(user, span_warning("A color that dark on an object like this? Surely not..."))
+				balloon_alert(user, "a color that dark on this? surely not.")
```

Comment:
balloon alerts shouldn't end in a single period and this is way too long anyways... it's clever but unfortunately a to_chat is  probably the way to go in this context